### DISCUSSION TITLE
Merge joins

### DISCUSSION THREAD

**Discussion by D063520** [eclipse-rdf4j/rdf4j#4220] [Q&A]
Hi,

as far as I know when implementing a new store in RDF4J we basically specify how to resolve a triple pattern by extending the class TripleSource and overwriting the method getStatements. This specify how to query for one triple pattern. As a consequence all joints are executed as nested joints.

Is it possible to specify how to execute merge joins? Is this part of the RDF4J implementation? Or is this not possible since when asking over the native store, which is a B-tree you will not get ordered triples.

Many questions, I hope they help to explain my problem ; )

Would really help!
Thank you
D063520

**Comment by kenwenzel** [comment]
I'm not sure if this fully answers your question, but you may have a look at [JoinQueryEvaluationStep](https://github.com/eclipse/rdf4j/blob/112bc7ef3d405c7889a6114a451dc2706d622e01/core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/evaluationsteps/JoinQueryEvaluationStep.java#L31) created by [StrictEvaluationStrategy](https://github.com/eclipse/rdf4j/blob/b849630bfac9c84e0444390179409456da043b15/core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/StrictEvaluationStrategy.java#L461).

**Reply by D063520** [reply]
this is a good pointer, probably one has to implement a new Join algorithm and in this part say to use it ....
